plugins/node/opentelemetry-instrumentation-connect/src/instrumentation.ts

```diff
@@ -150,10 +155,17 @@ export class ConnectInstrumentation extends InstrumentationBase<Server> {
       }
 
       res.addListener('close', finishSpan);
-      arguments[2] = instrumentation._patchNext(next, finishSpan);
+      arguments[nextArgIdx] = instrumentation._patchNext(next, finishSpan);
 
       return (middleWare as any).apply(this, arguments);
-    };
+    }
+
+    Object.defineProperty(patchedMiddleware, 'length', {
+      value: middleWare.length,
+      writable: false,
```

Comment:
Please add `configurable: true` here to avoid that anyone else who tries something similar fails in doing so.